polygeist/tools/cgeist/driver.cc

```diff
@@ -940,19 +978,29 @@ int main(int argc, char **argv) {
   mlir::MLIRContext context;
   registerDialects(context, syclKernelsOnly);
 
-  // Generate MLIR for the input files (in a single module).
-  mlir::OwningOpRef<mlir::ModuleOp> module(
-      mlir::ModuleOp::create(mlir::OpBuilder(&context).getUnknownLoc()));
+  // Generate MLIR for the input files.
+  auto modules = buildModules(context);
+  auto &module = modules.first;
```

Comment:
Using `auto` makes the code harder to read. Looking at these declarations I do not know the static type of `module` ... In general LLVM coding guidelines states that when the static type is not clear from the context one should refrain from using `auto`.

Reply:
Also, might be more clear to avoid building the 2 modules as a pair in the first place.

Reply:
Addressed. More readable now.